crates/pallet-dummy-precompiles-code/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+//! A substrate pallet.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::StorageVersion;
+pub use pallet::*;
+use sp_core::H160;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+/// The dummy code used to be stored for precompiles addresses - 0x5F5FFD (as raw bytes).
+///
+/// This is actually a hand-crafted sequence of opcodes for a bare-bones revert.
+/// The REVERT opcode (which is FD) - it takes two arguments from the stack with PUSH0 (5F twice).
+pub const DUMMY_CODE: [u8; 3] = [95, 95, 253];
+
+// We have to temporarily allow some clippy lints. Later on we'll send patches to substrate to
+// fix them at their end.
+#[allow(clippy::missing_docs_in_private_items)]
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{pallet_prelude::*, sp_std::vec::Vec};
+    use frame_system::pallet_prelude::*;
+
+    use super::*;
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    /// Configuration trait of this pallet.
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_evm::Config {
+        /// The current precompiles addresses creation version.
+        type CreationVersion: Get<u16>;
+
+        /// The list of precompiles adresses to be created at evm with dummy code.
+        type PrecompilesAddresses: Get<Vec<H160>>;
+    }
+
+    /// The creation version.
+    #[pallet::storage]
+    #[pallet::getter(fn creation_version)]
+    pub type CreationVersion<T: Config> = StorageValue<_, u16, ValueQuery>;
+
+    #[pallet::genesis_config]
+    #[derive(Default)]
+    pub struct GenesisConfig {}
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            for precompile_address in &T::PrecompilesAddresses::get() {
+                pallet_evm::Pallet::<T>::create_account(*precompile_address, DUMMY_CODE.to_vec());
+            }
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> Weight {
+            let creation_version = Self::creation_version();
+            let mut weight = T::DbWeight::get().reads(1);
+
+            if creation_version != T::CreationVersion::get() {
+                for precompile_address in &T::PrecompilesAddresses::get() {
+                    let code = pallet_evm::Pallet::<T>::account_codes(*precompile_address);
+                    weight += T::DbWeight::get().reads(1);
+
+                    if code != DUMMY_CODE.to_vec() {
```

Comment:
```suggestion
                    if &code != &DUMMY_CODE {
```

Try avoiding needless `Vec`s to save on `malloc`s.

Reply:
https://github.com/humanode-network/humanode/pull/777/commits/f0184dd0a7f4c37c675624de581f92f171486a5b